model/src/main/java/org/mskcc/cmo/metadb/model/MetadbSample.java

```diff
@@ -98,15 +98,18 @@ public void addSampleMetadata(SampleMetadata sampleMetadata) {
 
     /**
      *
-     * @return SampleIgoId
+     * @return SamplePrimaryId
      */
-    public SampleAlias getSampleIgoId() {
+    public String getSamplePrimaryId() {
         if (sampleAliases == null) {
             sampleAliases = new ArrayList<>();
         }
         for (SampleAlias s: sampleAliases) {
             if (s.getNamespace().equalsIgnoreCase("igoId")) {
-                return s;
```

Comment:
so primaryId is a member of `SampleMetadata` so I think what we want to do here instead is to be able to find a `SampleAlias` that matches the `primaryId`

You would be able to do something like:
```
SampleMetadata latestMetadata = getLatestSampleMetadata();
for (SampleAlias s: sampleAliases) {
  if (s.getValue().equalsIgnoreCase(latestMetadata.getPrimaryId()) {
    return s;
  }
}
```

Reply:
I would also rename the method to `getPrimarySampleAlias()` maybe? open to suggestions